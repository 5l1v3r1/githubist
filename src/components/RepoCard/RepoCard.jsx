// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import styles from './RepoCard.scss';

type Props = {
  slug: string,
  description?: string,
  language: {
    name: string,
    slug: string,
  },
  stars: number,
  forks: number,
};

const RepoCard = ({ slug, description, language, stars, forks }: Props) => (
  <div className={styles.repoCard}>
    <Link to={`/repo/${slug}`} className={styles.name}>
      {slug}
    </Link>
    {description && <p className={styles.description}>{description}</p>}

    <div className={styles.meta}>
      <Link to={`/language/${language.slug}`} className={cx(styles.metaItem, styles.language)}>
        {language.name}
      </Link>
      <span className={styles.metaItem}>{stars.toLocaleString()} Star</span>
      <span className={styles.metaItem}>{forks.toLocaleString()} Fork</span>
    </div>
  </div>
);

RepoCard.defaultProps = {
  description: '',
};

export default RepoCard;
