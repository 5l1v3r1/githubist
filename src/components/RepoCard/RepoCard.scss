@import '../../theme/theme';

.name {
  color: $black-color;
  font-weight: 600;
  text-decoration: none;
  word-break: break-all;

  &:hover {
    text-decoration: underline;
  }
}

.description {
  margin: 0;
  word-break: break-word;
}

.meta {
  display: flex;
  flex-direction: row;
  margin: map-get($spacers, normal) 0 0 0;
}

.metaItem {
  margin-right: map-get($spacers, normal);
  font-size: map-get($font-sizes, small);
  font-weight: 600;
}

.language {
  color: $secondary-color;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}

.githubCreatedAt {
  margin-right: 0;
  margin-left: auto;
}

@media only screen and (max-width: px-to-rem(1000px)) {
  .meta {
    flex-direction: column;
  }

  .metaItem {
    margin-right: 0;
  }
}
