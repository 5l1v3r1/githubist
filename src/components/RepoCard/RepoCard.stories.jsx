// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import backgrounds from '@storybook/addon-backgrounds';
import { MemoryRouter } from 'react-router-dom';

import RepoCard from './RepoCard';

storiesOf('RepoCard', module)
  .addDecorator(backgrounds([{ name: 'light', value: '#fcfafa', default: true }]))
  .add('default', () => (
    <MemoryRouter>
      <div style={{ width: '600px', margin: '16px' }}>
        <RepoCard
          slug="alpcanaydin/github.ist"
          description="Github Türkiye istatistikleri 2019"
          language={{ name: 'JavaScript', slug: 'javascript' }}
          stars={200}
          forks={30}
        />
      </div>
    </MemoryRouter>
  ))
  .add('with rank', () => (
    <MemoryRouter>
      <div style={{ width: '600px', margin: '16px' }}>
        <RepoCard
          rank={1}
          slug="alpcanaydin/github.ist"
          description="Github Türkiye istatistikleri 2019"
          language={{ name: 'JavaScript', slug: 'javascript' }}
          stars={200}
          forks={30}
        />
      </div>
    </MemoryRouter>
  ))
  .add('with github creation date', () => (
    <MemoryRouter>
      <div style={{ width: '600px', margin: '16px' }}>
        <RepoCard
          slug="alpcanaydin/github.ist"
          description="Github Türkiye istatistikleri 2019"
          language={{ name: 'JavaScript', slug: 'javascript' }}
          stars={200}
          forks={30}
          githubCreatedAt="2014-03-24T13:12:26.000000Z"
        />
      </div>
    </MemoryRouter>
  ))
  .add('without description', () => (
    <MemoryRouter>
      <div style={{ width: '600px', margin: '16px' }}>
        <RepoCard
          slug="alpcanaydin/github.ist"
          language={{ name: 'JavaScript', slug: 'javascript' }}
          stars={200}
          forks={30}
        />
      </div>
    </MemoryRouter>
  ));
