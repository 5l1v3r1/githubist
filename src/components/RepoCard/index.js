// @flow

import RepoCard from './RepoCard';

export default RepoCard;
