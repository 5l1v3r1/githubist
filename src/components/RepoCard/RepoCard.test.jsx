// @flow

import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import RepoCard from './RepoCard';

it('renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <RepoCard
        slug="alpcanaydin/github.ist"
        description="Github TÃ¼rkiye istatistikleri 2018"
        language={{ name: 'JavaScript', slug: 'javascript' }}
        stars={200}
        forks={30}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly without description', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <RepoCard
        slug="alpcanaydin/github.ist"
        language={{ name: 'JavaScript', slug: 'javascript' }}
        stars={200}
        forks={30}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});
