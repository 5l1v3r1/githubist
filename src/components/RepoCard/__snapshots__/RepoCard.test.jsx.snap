// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RepoCard
      description="Github Türkiye istatistikleri 2018"
      forks={30}
      language={
        Object {
          "name": "JavaScript",
          "slug": "javascript",
        }
      }
      slug="alpcanaydin/github.ist"
      stars={200}
    >
      <div>
        <Link
          replace={false}
          to="/repo/alpcanaydin/github.ist"
        >
          <a
            href="/repo/alpcanaydin/github.ist"
            onClick={[Function]}
          >
            alpcanaydin/github.ist
          </a>
        </Link>
        <p>
          Github Türkiye istatistikleri 2018
        </p>
        <div>
          <Link
            className=""
            replace={false}
            to="/language/javascript"
          >
            <a
              className=""
              href="/language/javascript"
              onClick={[Function]}
            >
              JavaScript
            </a>
          </Link>
          <span>
            200
             Star
          </span>
          <span>
            30
             Fork
          </span>
        </div>
      </div>
    </RepoCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly without description 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RepoCard
      description=""
      forks={30}
      language={
        Object {
          "name": "JavaScript",
          "slug": "javascript",
        }
      }
      slug="alpcanaydin/github.ist"
      stars={200}
    >
      <div>
        <Link
          replace={false}
          to="/repo/alpcanaydin/github.ist"
        >
          <a
            href="/repo/alpcanaydin/github.ist"
            onClick={[Function]}
          >
            alpcanaydin/github.ist
          </a>
        </Link>
        <div>
          <Link
            className=""
            replace={false}
            to="/language/javascript"
          >
            <a
              className=""
              href="/language/javascript"
              onClick={[Function]}
            >
              JavaScript
            </a>
          </Link>
          <span>
            200
             Star
          </span>
          <span>
            30
             Fork
          </span>
        </div>
      </div>
    </RepoCard>
  </Router>
</MemoryRouter>
`;
